NvimView/NvimView/NvimView+Draw.swift

```diff
@@ -96,14 +84,58 @@ extension NvimView {
       self.log.error("\(cursorRegion) vs. \(self.ugrid.size)")
       return
     }
-    guard let cursorAttrs = self.cellAttributesCollection.attributes(
+    guard let cellAtCursorAttrs = self.cellAttributesCollection.attributes(
       of: self.ugrid.cells[cursorPosition.row][cursorPosition.column].attrId
-    )?.reversed else {
+    ) else {
       self.log.error("Could not get the attributes" +
                      " at cursor: \(cursorPosition)")
       return
     }
 
+    guard self.modeInfoList.count > self.mode.rawValue else {
+      self.log.error("Could not get modeInfo for mode index \(self.mode.rawValue)")
+      return
+    }
+    let modeInfo = modeInfoList[Int(mode.rawValue)]
+
+    guard let cursorAttrId = modeInfo.attrId,
+      let cursorShapeAttrs = self.cellAttributesCollection.attributes(
+        of: cursorAttrId,
+        withDefaults: cellAtCursorAttrs
+      ) else {
+        self.log.error("Could not get the attributes" +
+          " for cursor in mode: \(mode) \(modeInfo)")
+        return
+    }
+
+    // will be used for clipping
+    var cursorRect: CGRect
+    var cursorTextColor: Int
```

Comment:
This can be `let`.

Reply:
Done, converted `cursorTextColor` to `let`. As `cursorRect` is modified later (either width or height, depending on cursor shape), it's not easily possible there without making a temporary copy and then assigning that. (Which I think is not worth it.)